recipes-benchmark/sel4-collect-traces/files/collect-traces.sh

```diff
@@ -0,0 +1,96 @@
+#!/bin/sh
+
+function usage () {
+     echo "Usage: $(basename $0) [-f 'funcs to trace'] [-t sec] [-c]"
+     exit 1
+}
+
+while getopts ':cf:t:h' opt; do
+  case "$opt" in
+    f)
+      func="$OPTARG"
+      ;;
+
+    c)
+      cat /sys/kernel/debug/tracing/available_filter_functions
+      exit 0
+      ;;
+
+    t)
+      time="$OPTARG"
+      ;;
+
+    h)
+      usage
+      ;;
+
+    :)
+      echo -e "option requires an argument."
+      usage
+      ;;
+
+    ?)
+      echo -e "Invalid command option."
+      usage
+      ;;
+  esac
+done
+shift "$(($OPTIND -1))"
+
+if [ -z "$func" ]
+then
+    usage
+fi
+
+if [ -z "$time" ]
+then
+    usage
+fi
+
+echo "start collecting data for following functions: ${func} for ${time} sec"
+
+echo 'Stop ftrace'
+echo 0 > /sys/kernel/debug/tracing/tracing_on
+
+echo 'Set PMU as clock source'
+echo pmu_clock > /sys/kernel/debug/tracing/trace_clock
+
+echo 'Configure ftrace'
+echo ${func} > /sys/kernel/debug/tracing/set_ftrace_filter
+echo function > /sys/kernel/debug/tracing/current_tracer
+
+echo 'Start ftrace'
+echo 1 > /sys/kernel/debug/tracing/tracing_on
+
+echo 'Start seL4 trace'
+echo 1 > /sys/kernel/debug/sel4_tracebuffer/trace_on
+
+echo Wait ${time}
+sleep ${time}
+
+echo 'Start seL4 trace'
+echo 0 > /sys/kernel/debug/sel4_tracebuffer/trace_on
+
+echo 'Stop ftrace'                                        
```

Comment:
Looks like this line has trailing whitespace. Check that you have coloring turned on: `git config --global color.ui auto`. Now when you do `git diff` (with or without `--staged`) you will see whitespace errors in red. For us that the colors on, whitespace errors are a bit annoying since they tend to steer our focus from what's actually changed.

Reply:
Sure. Fixed.